src/wire/dhcpv4.rs

```diff
@@ -98,6 +205,31 @@ impl<'a> DhcpOption<'a> {
                     (field::OPT_REQUESTED_IP, 4) => {
                         option = DhcpOption::RequestedIp(Ipv4Address::from_bytes(data));
                     }
+                    (field::OPT_TIME_OFFSET, 4) => {
+                        option = DhcpOption::TimeOffset(u32::from_be_bytes([
+                            data[0], data[1], data[2], data[3],
+                        ]));
+                    }
+                    (field::OPT_CLIENT_ARCH, _) => option = DhcpOption::ClientArchTypeList(data),
+                    (field::OPT_CLIENT_MACHINE_ID, _) => {
+                        let id_type = MachineIdType::try_from(data[0])?;
```

Comment:
This panics if `data.len() = 0`.

Reply:
@Dirbaio Um I don't seem to get the fuzzer to work. I tried running `cargo install cargo-fuzz` &&  `cargo fuzz run dhcp_header` 
``` 
Finished release [optimized] target(s) in 0.01s
     Running `target/x86_64-unknown-linux-gnu/release/dhcp_header -artifact_prefix=/home/lhebendanz/Projects/smoltcp/fuzz/artifacts/dhcp_header/ /home/lhebendanz/Projects/smoltcp/fuzz/corpus/dhcp_header`
INFO: Running with entropic power schedule (0xFF, 100).
INFO: Seed: 3421684815
INFO:        0 files found in /home/lhebendanz/Projects/smoltcp/fuzz/corpus/dhcp_header
INFO: -max_len is not provided; libFuzzer will not generate inputs larger than 4096 bytes
INFO: A corpus is not provided, starting from an empty corpus
#2	INITED exec/s: 0 rss: 31Mb
ERROR: no interesting inputs were found. Is the code instrumented for coverage? Exiting.
────────────────────────────────────────────────────────────────────────────────

Error: Fuzz target exited with exit status: 1
```

with rustc `nightly-2021-09-19` `clang --version`
```
clang version 13.0.0
Target: x86_64-unknown-linux-gnu
Thread model: posix
```

Reply:
Ah I had to update my nightly version

Reply:
Fixed